Window write support for SSD1673

vanwinkeljan opened the pull request:
Added support for writing to a window instead of the complete display

vanwinkeljan committed:
cfb: Set dimensions in character framebuffer

Set width, height and pitch dimensions in character framebuffer before
calling display write API.

Signed-off-by: Jan Van Winkel <jan.van_winkel@dxplore.eu>

vanwinkeljan committed:
drivers: Allow to set supported pix format in ssd drivers

Allow to set supported pixel format in ssd1306 & ssd1673 drivers

Signed-off-by: Jan Van Winkel <jan.van_winkel@dxplore.eu>

vanwinkeljan committed:
drivers: Added missing API functions in ssd drivers

Added missing API function set_orientation to ssd1306 & ssd1673
drivers.

Signed-off-by: Jan Van Winkel <jan.van_winkel@dxplore.eu>

vanwinkeljan committed:
samples: cfb: Moved overlay config files to boards dir

Moved the overlay configuration files to board directory.

Signed-off-by: Jan Van Winkel <jan.van_winkel@dxplore.eu>

vanwinkeljan committed:
samples: doc: Added README.rst for CFB sample

Added missing README.rst for character frame buffer sample

Signed-off-by: Jan Van Winkel <jan.van_winkel@dxplore.eu>

vanwinkeljan commented:
Addressed doc changes based on @dbkinder comments on PR #9149

dbkinder reviewed (approved):
+1 for doc changes

vanwinkeljan commented:
@jfischer-phytec-iot any input?

vanwinkeljan commented on include/display.h:
```diff
@@ -48,6 +48,10 @@ enum display_screen_info {
 	 */
 	SCREEN_INFO_EPD			= BIT(2),
+	/**
+	 * Screen has two alternating ram buffers
```
No currently there is no ```display_flush``` function available in the display API.

But ```SCREEN_INFO_DOUBLE_BUFFER``` is jut an indication to the higher layers, e.g. incase only a partial update of the display buffer is performed the same partial update needs to happen on the second buffer to keep the display content consistent.

jfischer-phytec-iot commented on drivers/display/ssd1673.c:
```diff
@@ -415,49 +426,74 @@ static int ssd1673_controller_init(struct device *dev)
-	if (ssd1673_write_cmd(driver, SSD1673_CMD_SW_RESET, NULL, 0)) {
-		return -1;
+	err = ssd1673_write_cmd(driver, SSD1673_CMD_SW_RESET, NULL, 0);
+	if (err < 0) {
```
Please stop to add it, it does not really help and adds unnecessary code

jfischer-phytec-iot commented on drivers/display/ssd1673.c:
```diff
@@ -169,54 +173,32 @@ static int ssd1673_suspend(const struct device *dev)
 				 &tmp, sizeof(tmp));
 }
 
-static int ssd1673_update_display(const struct device *dev, bool initial)
```
Why are you revise this funktion? Why this change is necessary for region update?

vanwinkeljan replied:
parameter is no longer needed as init is done during display init

jfischer-phytec-iot commented on drivers/display/ssd1673.c:
```diff
@@ -363,11 +320,8 @@ static int ssd1673_set_brightness(const struct device *dev,
 
 static int ssd1673_set_contrast(const struct device *dev, u8_t contrast)
```
It was a nice hack for initial update, why are you removing it? and why in this commit?

vanwinkeljan replied:
It was no longer needed as the initial clear is done at init

jfischer-phytec-iot replied:
This commit must be reverted because it removed possibility for the user to perform initial update.

jfischer-phytec-iot commented on drivers/display/ssd1673.c:
```diff
@@ -225,118 +207,93 @@ static int ssd1673_write(const struct device *dev, const u16_t x,
+		return err;
 	}
 
-	/* clear unusable page */
```
That must stay, you can not really use these lines.

vanwinkeljan replied:
There is no need to clear this lines on every write as it is done once on init

jfischer-no reviewed (dismissed):
Please split 171c4f440d392618e905eec78d4762ae457476c5 in logical peaces and remove unnecessary code changes, keep the driver design and structure.

vanwinkeljan committed:
drivers: Return correct erorr codes in SSD1673 driver

Return correct error codes in SSD1673 driver instead of -1

Signed-off-by: Jan Van Winkel <jan.van_winkel@dxplore.eu>

vanwinkeljan committed:
drivers: Clear SSD1673 display during display init

Clear SSD1673 display during display driver initialization instead of
triggering clear via setting contrast.

Signed-off-by: Jan Van Winkel <jan.van_winkel@dxplore.eu>

vanwinkeljan committed:
drivers: display: Added windowing for SSD1673

Added support to the SSD1673 dipslay for partially writing the display

Signed-off-by: Jan Van Winkel <jan.van_winkel@dxplore.eu>

vanwinkeljan commented on include/display.h:
```diff
@@ -48,6 +48,10 @@ enum display_screen_info {
+	/**
+	 * Screen has two alternating ram buffers
+	 */
+	SCREEN_INFO_DOUBLE_BUFFER	= BIT(3),
```
moved to #6826

vanwinkeljan commented:
@jfischer-phytec-iot Splitted commits as requested and provided replies where needed

nashif commented:
@jfischer-phytec-iot feedback please

vanwinkeljan commented:
@jfischer-phytec-iot any further input?
